Entity

Maksim-maevsky opened the pull request.

ShirshovAlex committed:
Entity part two (#3)

* Internship entity

* Internship entity

Maksim-maevsky committed:
created entity Form and Interview (#4)

ShirshovAlex committed:
entity part three (#5)

Co-authored-by: Olga <olgaaevdokim@gmail.com>

Maksim-maevsky committed:
uncomented variables

Maksim-maevsky committed:
changed LocalDate instead Date , created InternshipLocation

Maksim-maevsky committed:
added InternshipLocation to Internship

Maksim-maevsky committed:
added UUID? and changed variable PATH in DockerFile like path

Maksim-maevsky committed:
changed int

Maksim-maevsky committed:
added timeSlot

ShirshovAlex committed:
change enum timeslot

ShirshovAlex committed:
change enum timeslot

Maksim-maevsky committed:
added TimeForCall

kzaretskaya commented on pom.xml:
```diff
@@ -1,56 +1,66 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
-	xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 https://maven.apache.org/xsd/maven-4.0.0.xsd">
```
Синхронизируйтесь, пожалуйста, друг с другом насчет code formatting (4 spaces или 2 tabs)
![image](https://user-images.githubusercontent.com/77664426/112844727-bc213680-90ac-11eb-9d3e-aa3c6bdaba23.png)

Разное форматирование мешает отслеживать реальные изменения в коде.
Optional создать файл с code style

Maksim-maevsky replied:
everybody has 4 spaces format, we need to read about code style file, i am going create task in Trella about it.

kzaretskaya commented on pom.xml:
```diff
@@ -1,56 +1,66 @@
-					<artifactId>junit-vintage-engine</artifactId>
-				</exclusion>
-			</exclusions>
+			<artifactId>spring-boot-starter-web</artifactId>
```
Версия?

kzaretskaya commented on pom.xml:
```diff
@@ -1,56 +1,66 @@
-		</dependency>
+        <dependency>
+            <groupId>org.springframework.boot</groupId>
+            <artifactId>spring-boot-starter-test</artifactId>
```
The same

Maksim-maevsky replied:
![image](https://user-images.githubusercontent.com/65280608/112851380-99465080-90b3-11eb-8754-ce0bdb7b3667.png)

kzaretskaya replied:
Good practice to specify it directly. If your dependencies `spring-boot-starter-*`  use the same version as parent there can be separated to common property. Example:
![image](https://user-images.githubusercontent.com/77664426/112856280-4a4eea00-90b8-11eb-8d40-8a86a198d3cd.png)
![image](https://user-images.githubusercontent.com/77664426/112856361-62266e00-90b8-11eb-8841-1edaecb4f7ac.png)

Maksim-maevsky replied:
done

kzaretskaya commented on src/main/java/by/exadel/internship/team_four/entity/Internship.java:
```diff
@@ -0,0 +1,54 @@
+    private LocalDate startDateRegistration;
+    private LocalDate endDateRegistration;
+
+    public Internship(String name, LocalDate startDateInternship, LocalDate endDateInternship,
```
Зачем здесь конструктор если вверху есть аннотация @AllArgsConstructor ?

Maksim-maevsky replied:
this constructor without id

kzaretskaya replied:
For which purposes do you need constructor without id? Probably some fields can also be empty. Suggestion: user @Builder

Maksim-maevsky replied:
done

kzaretskaya commented on src/main/java/by/exadel/internship/team_four/entity/Internship.java:
```diff
@@ -0,0 +1,54 @@
+
+    private UUID id;
+    private String name;
+    private LocalDate startDateInternship;
```
```suggestion
    private LocalDate startDate;
```

kzaretskaya commented on src/main/java/by/exadel/internship/team_four/entity/Internship.java:
```diff
@@ -0,0 +1,54 @@
+    private UUID id;
+    private String name;
+    private LocalDate startDateInternship;
+    private LocalDate endDateInternship;
```
Это поле уже является частью класса Internship.
```suggestion
    private LocalDate endDate;
```

kzaretskaya commented on src/main/java/by/exadel/internship/team_four/entity/enums/UserRole.java:
```diff
@@ -0,0 +1,7 @@
+package by.exadel.internship.team_four.entity.enums;
+
+public enum UserRole {
```
Без префикса ROLE_

Maksim-maevsky replied:
done

kzaretskaya commented on src/main/java/by/exadel/internship/team_four/entity/Form.java:
```diff
@@ -0,0 +1,42 @@
+package by.exadel.internship.team_four.entity;
```
Достаточно названия приложения
```suggestion
package by.exadel.internship.entity;
```

Maksim-maevsky replied:
done

kzaretskaya commented on src/main/java/by/exadel/internship/team_four/entity/enums/EnglishLevel.java:
```diff
@@ -0,0 +1,10 @@
+package by.exadel.internship.team_four.entity.enums;
+
+  public enum EnglishLevel {
+    LEVEL_A1,
```
Excessive prefix `LEVEL_`

Maksim-maevsky replied:
done

kzaretskaya commented on src/main/java/by/exadel/internship/team_four/entity/enums/FormStatus.java:
```diff
@@ -0,0 +1,18 @@
+    REJECTED,         // отклонен (не соответствие требованиям (тех знания))
+    CANDIDATE_REFUSED,   // отказался (сам кандидат)
+
+    WAITING, //спросить
```
Save these statuses to some documentation and remove now till real implementation

Maksim-maevsky replied:
is it good practice, if we dont use it now, we should keep them to some documentation? because we will use them for first version our project (second resources)? it is not for future versions

kzaretskaya replied:
As far as I know business logic implementation  is postponed for next statuses WAITING, GOT_JOB_OFFER and DIDNT_GOT_JOB_OFFER. I said only about these three statuses above.

Maksim-maevsky replied:
done

kzaretskaya commented on src/main/java/by/exadel/internship/team_four/entity/User.java:
```diff
@@ -0,0 +1,28 @@
+
+    private UUID id;
+    private String lastName;
+    private String name;
```
```suggestion
    private String firstName;
```

Maksim-maevsky replied:
done  
![image](https://user-images.githubusercontent.com/65280608/112868219-7a03ef00-90c4-11eb-87e4-90ae71bfe80a.png)

Maksim-maevsky committed:
removed Role_ prefix for user status, removed package team_four, removed Internship word from name of date variable.

kzaretskaya commented on src/main/java/by/exadel/internship/team_four/entity/Form.java:
```diff
@@ -0,0 +1,42 @@
+    private String filePath;
+    private Interview interview;
+    private FormStatus formStatus;
+    private TimeForCall timeForCall;
```
Is it approved that UI send you only one time period? Or it's questionable?

Maksim-maevsky replied:
we would like to  ask Zanna about it ? but we hadnt meteeng today, and we are going to write her our questions. Now we decided to leave this one

kzaretskaya replied:
Please add comment that this field is questionable.

Maksim-maevsky replied:
done

kzaretskaya commented on src/main/java/by/exadel/internship/team_four/entity/enums/FormatOfInternship.java:
```diff
@@ -0,0 +1,6 @@
+package by.exadel.internship.team_four.entity.enums;
+
+public enum FormatOfInternship {
```
Please, replace everywhere you use genitive nouns

```suggestion
public enum InternshipFormat {
```

kzaretskaya replied:
It's not resolved

Maksim-maevsky replied:
sorry, it's done now

kzaretskaya commented on src/main/java/by/exadel/internship/team_four/entity/Interview.java:
```diff
@@ -0,0 +1,24 @@
+    private UUID id;
+    private User admin;
+    private LocalDateTime adminInterviewDate;
+    private String feedback;
```
May be, better specify like adminFeedback if you've already specified it for techSpecialist below.

Maksim-maevsky replied:
done
![image](https://user-images.githubusercontent.com/65280608/112868305-930ca000-90c4-11eb-8479-7890830fa0a9.png)

kzaretskaya commented on src/main/java/by/exadel/internship/team_four/entity/InternshipLocation.java:
```diff
@@ -0,0 +1,17 @@
+@Data
+@AllArgsConstructor
+@NoArgsConstructor
+public class InternshipLocation {
```
@viktarkolbik what do you think?
```suggestion
public class Location {
```

Maksim-maevsky replied:
Form has fields country and city, and we used this name to understand this location for internship

kzaretskaya replied:
Okay, let's leave as it is now. In future if this class can be reused in other dto we can make generic this class to LocationReference and share among other dtos.

Maksim-maevsky replied:
done

Maksim-maevsky committed:
removed prefix Level for variables of Enum EnglishLevel, create property variables for versions in pom.xml etc

mortal-gpt commented on src/main/java/by/exadel/internship/dto/Internship.java:
```diff
@@ -0,0 +1,54 @@
+import java.util.UUID;
+
+@Data
+@AllArgsConstructor
```
the Constructor will have too much arguments, better to use @Builder or common  setters. This applies to all POJOs in your PR.

Maksim-maevsky replied:
done fore DTOs that have a lot of arguments

mortal-gpt commented on src/main/java/by/exadel/internship/dto/User.java:
```diff
@@ -0,0 +1,29 @@
+@Data
+@AllArgsConstructor
+@NoArgsConstructor
+public class User {
```
As far as I understand, your entities will have the same names, so It's good practice to add  'DTO' postfix to the class name. Please, do it for all DTOs.

Maksim-maevsky replied:
done

mortal-gpt commented on pom.xml:
```diff
@@ -1,56 +1,78 @@
+                <artifactId>spring-boot-maven-plugin</artifactId>
+                <version>${spring.version}</version>
+            </plugin>
+            <plugin>
```
What was the reason to add _maven-compiler-plugin_ plugin? Why source version - 8?

Maksim-maevsky replied:
changed targed and sources version, now it is 11.  Added for consistency, excluding default versions.

Maksim-maevsky committed:
changed name of class InternshipLocation? now it is Location.
create txt file doc.txt / removed  WAITING, GOT_JOB_OFFER, DIDNT_GOT_JOB_OFFER from RoleStatus and copy in doc.txt,

Maksim-maevsky committed:
InternshipFormat instead FormatOfInternship

Maksim-maevsky committed:
changed 11 version for maven plugin

kzaretskaya commented on src/main/java/by/exadel/internship/dto/FormDTO.java:
```diff
@@ -0,0 +1,41 @@
+    private FormStatus formStatus;
+    //this field is questionable
+    private TimeForCallDTO timeForCall;
+
```
Minor: remove excessive empty lines after fields

Maksim-maevsky replied:
![image](https://user-images.githubusercontent.com/65280608/112983924-e9cab600-9166-11eb-84c3-c30b93a68dfa.png)
done

kzaretskaya commented on src/main/java/by/exadel/internship/dto/FormDTO.java:
```diff
@@ -0,0 +1,41 @@
+    //not req
+    private String middleName;
+    private String email;
+    private String phoneNumb;
```
```suggestion
    private String phoneNumber;
```

Maksim-maevsky replied:
![image](https://user-images.githubusercontent.com/65280608/112984636-d1a76680-9167-11eb-94c7-e8eda8545242.png)
done

kzaretskaya commented on src/main/java/by/exadel/internship/dto/InternshipDTO.java:
```diff
@@ -0,0 +1,56 @@
+    private LocalDate startDateRegistration;
+    private LocalDate endDateRegistration;
+
+    public InternshipDTO(String name, LocalDate startDateInternship, LocalDate endDateInternship,
```
Remove please

Maksim-maevsky replied:
done

kzaretskaya commented on src/main/java/by/exadel/internship/dto/InternshipDTO.java:
```diff
@@ -0,0 +1,56 @@
+    private LocalDate publicationDate;
+    private String techSkills;
+    private List<LocationDTO> countryList;
+    private InternshipFormat formatOfInternship;
```
```suggestion
    private InternshipFormat internshipFormat;
```

Maksim-maevsky replied:
![image](https://user-images.githubusercontent.com/65280608/112984744-f00d6200-9167-11eb-851c-d43d2396deb9.png)
done

kzaretskaya commented on src/main/java/by/exadel/internship/dto/InternshipDTO.java:
```diff
@@ -0,0 +1,56 @@
+    private String description;
+    private String requirements;
+    private Integer capacity;
+    private LocalDate startDateRegistration;
```
```suggestion
    private LocalDate registrationStartDate;
```

for end date the same

Maksim-maevsky replied:
![image](https://user-images.githubusercontent.com/65280608/112984974-3ebafc00-9168-11eb-8162-9f561d3f2990.png)
done

kzaretskaya commented on src/main/java/by/exadel/internship/dto/UserDTO.java:
```diff
@@ -0,0 +1,31 @@
+    private List<InternshipDTO> listOfInternships;
+    private List<LocalDateTime> freeInterviewDates;
+
+
```
Minor: remove empty line

Maksim-maevsky replied:
![image](https://user-images.githubusercontent.com/65280608/112985173-86da1e80-9168-11eb-8eca-b60d024361d0.png)
done

kzaretskaya commented:
@Maksim-maevsky please, fix latest comments and PR will be ready to merge

Maksim-maevsky committed:
removed empty lines

Maksim-maevsky committed:
removed empty lines , and changed names

Maksim-maevsky committed:
removed doc.txt,  copy information to README.md

Maksim-maevsky committed:
Added A0 english level

kzaretskaya commented:
Thank you for effords! 💯
